consumers/consumer.go

```diff
@@ -4,8 +4,14 @@
 package consumers
 
 // Consumer specifies message consuming API.
-type Consumer interface {
-	// Consume method is used to consumed received messages.
-	// A non-nil error is returned to indicate operation failure.
-	Consume(messages interface{}) error
+type AsyncConsumer interface {
+	// Consume method is used to consume received messages.
+	// An error channel is used to handle errors, so it supports the async approach.
+	ConsumeAsync(messages interface{}, errs chan<- error)
```

Comment:
Consider adding the `Errors() <- chan error` method instead of receiving `errs` chan.